File: Challenges/Algorithms/Warmup/Mini-Max Sum/mini-max-sum.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    arr = readLine().split(' ');
    arr = arr.map(Number);

    var sum_min = 0;
    var sum_max = 0;
    arr.sort();
    for (var i = 0; i < arr.length - 1; i++) {
        sum_min += arr[i];
        sum_max += arr[i+1];
    }

    console.log(sum_min + " " + sum_max);
}

File: Challenges/Algorithms/Warmup/Mini-Max Sum/mini-max-sum.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    long sum_min = 0;
    long sum_max = 0;
    long number = 0;
    std::vector<long> numbers;
    while (std::cin >> number) {
        numbers.push_back(number);
    }

    std::sort(numbers.begin(), numbers.end());
    for (unsigned int i = 0; i < numbers.size() - 1; i++) {
        sum_min += numbers[i];
        sum_max += numbers[i+1];
    }

    std::cout << sum_min << " " << sum_max << std::endl;
    return 0;
}
